util/dir/src/lib.rs

```diff
@@ -226,16 +226,26 @@ impl DatabaseDirectories {
 	}
 }
 
+fn default_path(t: AppDataType, product_name: &'static str) -> Option<PathBuf> {
+	let app_info = AppInfo { name: product_name, author: AUTHOR };
+	let old_root = get_app_root(t, &app_info).ok()?;
+	if old_root.exists() {
+		return Some(old_root);
+	}
+
+	let mut root = data_root(t).ok()?;
+	root.push("OpenEthereum");
+	Some(root)
+}
+
 /// Default data path
 pub fn default_data_path() -> String {
-	let app_info = AppInfo { name: PRODUCT, author: AUTHOR };
-	get_app_root(AppDataType::UserData, &app_info).map(|p| p.to_string_lossy().into_owned()).unwrap_or_else(|_| "$HOME/.parity".to_owned())
+	default_path(AppDataType::UserData, PRODUCT).map(|p| p.to_string_lossy().into_owned()).unwrap_or_else(|| "$HOME/.openethereum".to_owned())
```

Comment:
Maybe I'm reading this wrong but it seems like you're going to end up with a path containing `io.parity.ethereum`/`Parity` on *nix/windows, and then `OpenEthereum` added to that. That isn't what we want I think?
When I run the test here it passes but printing it I see: `Directories { base: "/home/dvdplm/.local/share/io.parity.ethereum", db: "/home/dvdplm/.local/share/io.parity.ethereum/chains", cache: "/home/dvdplm/.local/share/io.parity.ethereum/cache", keys: "/home/dvdplm/.local/share/io.parity.ethereum/keys", signer: "/home/dvdplm/.local/share/io.parity.ethereum/signer", secretstore: "/home/dvdplm/.local/share/io.parity.ethereum/secretstore" }`

Reply:
I retained that code as the old path mechanism. If the old constructed path does not exist it is discarded. It probably prints the old path because you have it on your disk already?

Reply:
macOS

```
Directories { base: "/Users/vorot93/Library/Application Support/OpenEthereum", db: "/Users/vorot93/Library/Application Support/OpenEthereum/chains", cache: "/Users/vorot93/Library/Application Support/OpenEthereum/cache", keys: "/Users/vorot93/Library/Application Support/OpenEthereum/keys", signer: "/Users/vorot93/Library/Application Support/OpenEthereum/signer", secretstore: "/Users/vorot93/Library/Application Support/OpenEthereum/secretstore" }
```

if I create the legacy folder manually

```
Directories { base: "/Users/vorot93/Library/Application Support/io.parity.ethereum", db: "/Users/vorot93/Library/Application Support/io.parity.ethereum/chains", cache: "/Users/vorot93/Library/Application Support/io.parity.ethereum/cache", keys: "/Users/vorot93/Library/Application Support/io.parity.ethereum/keys", signer: "/Users/vorot93/Library/Application Support/io.parity.ethereum/signer", secretstore: "/Users/vorot93/Library/Application Support/io.parity.ethereum/secretstore" }
```

Reply:
>  It probably prints the old path because you have it on your disk already?

Ah, didn't realize the test was not sandboxed.

k, so maybe we can extend the unit tests to make sure we're doing the right thing on different platforms?